modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/h2/TableWithImplicitPkTest.java

```diff
@@ -0,0 +1,295 @@
+import static org.apache.ignite.testframework.GridTestUtils.assertThrows;
+
+/**
+ * Basic tests to check the possibility of creating tables without specifying a primary key.
+ */
+@WithSystemProperty(key = IGNITE_SQL_ALLOW_IMPLICIT_PK, value = "true")
+public class TableWithImplicitPkTest extends GridCommonAbstractTest {
+    /** Nodes count. */
+    private static final int NODES_CNT = 3;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        return super.getConfiguration(igniteInstanceName)
+            .setCacheConfiguration(cacheCfg())
+            .setDataStorageConfiguration(
+                new DataStorageConfiguration()
+                    .setDefaultDataRegionConfiguration(
+                        new DataRegionConfiguration()
+                            .setPersistenceEnabled(true)
+                    )
+            );
+    }
+
+    /**
+     * @return Cache configuration.
+     */
+    private CacheConfiguration<?, ?> cacheCfg() {
+        return new CacheConfiguration<>(DEFAULT_CACHE_NAME)
+            .setAffinity(new RendezvousAffinityFunction(false, 16))
+            .setBackups(1);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        cleanPersistenceDir();
+
+        startGridsMultiThreaded(NODES_CNT)
+            .addCacheConfiguration(cacheCfg());
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        List<String> cachesToDestroy = grid(0).cacheNames().stream()
+            .filter(name -> !DEFAULT_CACHE_NAME.equals(name))
+            .collect(Collectors.toList());
+
+        if (!cachesToDestroy.isEmpty()) {
+            grid(0).destroyCaches(cachesToDestroy);
+            awaitPartitionMapExchange();
+        }
+    }
+
+    @Test
+    public void testBasicOperations() {
+        String sql = "CREATE TABLE integers(i INTEGER)";
+
+        // "WRAP_KEY" option is not supported.
+        assertThrows(log, () -> querySql(sql + " WITH \"wrap_key=true\""), IgniteSQLException.class, null);
+
+        querySql(sql);
+
+        int rowsCnt = 5;
+        Set<Integer> expNums = new HashSet<>();
+
+        for (int n = 0; n < rowsCnt; n++) {
+            expNums.add(n);
+
+            querySql("INSERT INTO integers VALUES(?)", n);
+        }
+
+        List<List<?>> rows = querySql("SELECT * FROM integers");
+        assertEquals(rowsCnt, rows.size());
+
+        Set<Integer> resSet = new HashSet<>();
+
+        for (List<?> row : rows) {
+            assertEquals(1, row.size());
+
+            resSet.add((Integer)row.get(0));
+        }
+
+        assertEquals(expNums, resSet);
+    }
+
+    @Test
+    public void testCacheApiCompatibility() {
+        querySql("CREATE TABLE person(name VARCHAR, age INT) WITH \"value_type=" + Person.class.getName() + "\"");
```

Comment:
Let's also check the explain command if it is supported for DDL.
And system views.